### PR TITLE
To Implement a badge preview on the top of the screen

### LINKED ISSUE #14: Implement a badge preview on the top of the screen
There are a number of apps to use the phone as an LED screen. Please take inspiration and implement a preview screen to show how it will look on the actual device on the top. Compare for example to this app: https://play.google.com/store/apps/details?id=com.rooftopSpace.Led

### PR DESCRIPTION
**Pull request by adityastic**
Fixes #14 
Motive : Implement a badge preview on the top of the screen
Changes: 
- Created a Custom View (PreviewBadge.kt) for getting the complete visual of an Actual LED
- Added seperate data classes(Cell.kt, CheckList.kt) for better management
- Changes to MessageActivity to incorporate Support for Badge
- Vector Drawable's for LED
- Added Function in Presenter(MessagePresenter.kt) to take care of the hex conversions
- Changes to Message Activity XML

Screenshots of the changes:
Screenshot_20190325-144859_Badge Magic
Screenshot_20190325-144909_Badge Magic

### CODE REVIEW

**Comment by rishabhj126** [comment]
does this include the animations(marquee and flash )preview too?

**Comment by adityastic** [comment]
@rishabhj126 No Sir, This is just a preview for the Led Badge. I am planning to implement the animations in the next phase as it is stated as a different feature

File: app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+import android.graphics.drawable.Drawable
+import android.support.annotation.Nullable
+import android.util.AttributeSet
+import android.view.View
+import com.nilhcem.blenamebadge.R
+
+import java.math.BigInteger
+import java.util.ArrayList
+
+class PreviewBadge : View {
```
[the remark this answers is not in the document]
Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt]: TextView with a Font wont work for bitmaps or other custom hex strings

[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt]: Ok. There are a few optimizations needed. I have reviewed those

File: app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    private var mLed: Drawable? = null
+    private var mLedLit: Drawable? = null
```
[the remark this answers is not in the document]
Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt]: Alright Sir. Will change them

File: app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+    private lateinit var bgBounds: RectF
+    private var cells = ArrayList<Cell>()
+
+    private var checkList: ArrayList<CheckList>? = null
+
+    private fun resetCheckList() {
+        checkList = ArrayList()
+        for (i in 0..10) {
+            checkList!!.add(CheckList())
```
[the remark this answers is not in the document]
Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt]: Sure

File: app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt]: Should I force push the changes?

File: app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt]: @iamareebjamal Ive removed all the null assertions :)

**Comment by adityastic** [comment]
@iamareebjamal Check Now :), I've made the required changes

File: app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        val bgPaint = Paint()
+        bgPaint.isAntiAlias = true
+        bgPaint.color = Color.parseColor("#000000")
+
+        // Draw Background
+        canvas.drawRoundRect(bgBounds, 25f, 25f, bgPaint)
+
+        // Draw Cells
+        for (i in 0..10) {
+            for (j in 0..43) {
```
Comment (iamareebjamal) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt]: There are a lot of magic numbers

Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt]: On it

**Comment by adityastic** [comment]
@iamareebjamal Removed Magic Numbers and used until properly

**Pull request merged**